text/0000-hyphenation-data-api.md

```diff
@@ -0,0 +1,111 @@
+    hyphenation: true
+  }
+})
+```
+_We should consider making this behavior the default, as the spellChecker API is also enabled by default_
+
+This will enable hyphenation for the languages that are supported by the hyphenation data files, which is automatically downloaded.
+
+### Use of Google services
+
+Like the spellChecker API, hyphen-data files are downloaded from a Google CDN. If you want to avoid this you can provide an alternative URL to download the hyphen-data from: 
+```javascript
+myWindow.webContents.session.setHyphenationDataDownloadURL('https://example.com/hyphen-data/')
+```
+
+### Consideration: list supported languages
+
+The spellcheck API has a method to list the supported languages. We should consider adding a similar method to list the supported languages for hyphenation:
+
+```js
+// an array of all available languages for hyphenation
+const possibleLanguages = myWindow.webContents.session.availableHyphenationLanguages
+```
+ 
+## Reference-level explanation
+<!-- This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+- Any new dependencies on Chromium code are outlined. 
+
+The section should return to the examples given in the previous section, and explain more fully how
+the detailed proposal makes those examples work.
+
+-->
+
+Chromium requests hyphenation data via [`ContentBrowserClient::GetHyphenationDictionary()`](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/content_browser_client.h;l=2818;drc=b0423977b6ed86291838e90231b036b68784aae3). [`ChromeContentBrowserClient`'s override of this method](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_content_browser_client.cc;l=8168;drc=f667feb8a5c6f227b49328ce78a062acc4f81187) uses the [ComponentUpdater](https://source.chromium.org/chromium/chromium/src/+/main:components/component_updater/README.md) to download the data.
+
+In Electron, `GetHyphenationDictionary()` can be overridden by `ElectronBrowserClient`. This needs to download the hyphenation data and then run the callback passed to it with the directory containing the downloaded files.
+
+Downloading the hyphenation data is the difficult part. Electron doesn't use the Component Updater, and manually downloading and extracting the hyphenation data component seems suboptimal. However, the data could instead be downloaded from <https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/hyphenation-patterns/hyb> ([gz'ed path](https://chromium.googlesource.com/chromium/src/+archive/refs/heads/main/third_party/hyphenation-patterns/hyb.tar.gz), 1086 KiB). The component downloaded by Chromium has exactly the same contents as that directory. Alternatively, Electron could move the responsibility of downloading the hyphenation data to the user.
```

Comment:
We should follow what we do for spellcheck which is to download from Google CDN unless user specifies alternative url to download from.  Additionally, we should make the hyphenation data available as a build asset like we do with the spellcheck dictionary.